src/langchain_google_firestore/document_loader.py

```diff
@@ -0,0 +1,217 @@
+        """
+        db_batch = self.client.batch()
+
+        if document_ids and (len(document_ids) != len(documents)):
+            raise ValueError(
+                "`documents` and `document_ids` parameters must be the same length"
+            )
+
+        docs_list = itertools.zip_longest(documents, document_ids or [])
+
+        for batch in more_itertools.chunked(docs_list, WRITE_BATCH_SIZE):
+            for doc, doc_id in batch:
+                document_dict = DocumentConverter.convert_langchain_document(
+                    doc, self.client
+                )
+                if self.collection:
+                    doc_ref = self.client.collection(self.collection).document()
+                elif doc_id:
+                    doc_ref = DocumentReference(*doc_id.split("/"), client=self.client)
+                elif (
+                    ("reference" in document_dict)
+                    and ("type" in document_dict["reference"])
+                    and (document_dict["reference"]["type"] == TYPE)
+                    and isinstance(document_dict["reference"]["path"], str)
+                ):
+                    print("debug")
+                    print(document_dict)
+                    doc_ref = DocumentReference(
+                        *document_dict["reference"]["path"].split("/"),
+                        client=self.client,
+                    )
+                else:
+                    continue
+
+                db_batch.set(
+                    reference=doc_ref, document_data=document_dict["data"], merge=merge
+                )
+            db_batch.commit()
+
+    def delete_documents(
+        self, documents: List[Document], document_ids: Optional[List[str]] = None
+    ) -> None:
+        """Delete documents from the Firestore database.
+        Args:
+          documents: List of documents to be deleted from Firestore. It will try to extract
+            the {document_path} from the `reference` in the document metadata.
+          document_ids: List of documents ids to be deleted from Firestore. If provided
+            the `documents` argument will be ignored.
+
+        """
+        db_batch = self.client.batch()
+
+        docs_list = itertools.zip_longest(documents, document_ids or [])
+
+        for batch in more_itertools.chunked(docs_list, WRITE_BATCH_SIZE):
+            for doc, doc_id in batch:
+                document_path = None
+                if doc_id:
+                    document_path = doc_id
+                elif doc:
+                    document_dict = DocumentConverter.convert_langchain_document(
+                        doc, self.client
+                    )
+                    if (
+                        ("reference" in document_dict)
+                        and ("type" in document_dict["reference"])
+                        and (document_dict["reference"]["type"] == TYPE)
+                    ):
+                        document_path = document_dict["reference"]["path"]
```

Comment:
1. fail if path is missing in document
2. add clarification that we can only delete with doc path